### DISCUSSION TITLE
Why force kid with public prefix?

### DISCUSSION THREAD

**Discussion by fjvierap** [ory/hydra#2435] [Q&A] [answered]
Hi, what is the motivation of forcing keys to be exposed in .well-known/jwks.json to be prefixed with public?

Thanks in advance

**Comment by fjvierap** [comment, marked as answer]
I guess that is for not display private ones... is there some way to do it without forcing the public, prefix? Maybe instead of forcing public hide private ones?

**Reply by aeneasr** [reply]
The code is rather old - the original idea was to not expose private keys but we could probably update the code to not use these prefixes.
